modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointDirtyPages.java

```diff
@@ -250,7 +215,7 @@ private CheckpointDirtyPages owner() {
         }
 
         private boolean needsNextRegion() {
-            return toPosition == dirtyPages.get(regionIndex).dirtyPages.pageIds.length;
+            return toPosition == dirtyPages.get(regionIndex).dirtyPages.length;
```

Comment:
You decided to leave this extra lookup, I see

Reply:
Yes, the **regionIndex** is needed to search for the **getNextPartitionView**

Reply:
But for what? according to me, you need to add only what you need, do you insist that it be?

Reply:
I already explained myself. `dirtyPages.get(regionIndex).dirtyPages` - this extra lookup is unnecessary. You're calculating the same object every time.

Reply:
I would not like to add a field.